src/client/components/Modal/Modal.component.js

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable no-param-reassign */
+import React from 'react';
+import './Modal.styles.css';
+import PropTypes from 'prop-types';
+import Button from '../Button/Button.component';
+import addedToCart from '../../assets/images/added_to_cart.png';
+
+export default function Modal({
+  modalIsOpen,
+  message,
+  cornerCloseButton,
+  primaryButtonLabel,
+  showConfirmIcon,
+  buttonFunction,
+  secondaryButtonFunction,
+  secondaryButtonLabel,
+}) {
+  return (
+    <div className="modal-background">
+      <div className="modal-box">
+        {cornerCloseButton && (
+          <button
+            onClick={() => modalIsOpen(false)}
+            type="button"
+            className="closing-modal-button"
+          >
+            X
+          </button>
+        )}
+        <p className="modal-text">{message}</p>
+        {showConfirmIcon && (
+          <img
+            src={addedToCart}
+            alt="add to cart approved"
+            style={{ marginBottom: '25px', marginTop: '25px', width: '2.5rem' }}
+          />
+        )}
+        <div className="button-container">
+          <Button
+            primary={true}
+            label={primaryButtonLabel}
+            onClick={buttonFunction}
+            style={{
+              marginBottom: '40px',
+              marginTop: '30px',
+              width: 'fit-content',
+              borderRadius: '4px',
+            }}
+          />
+          {secondaryButtonFunction && secondaryButtonLabel && (
+            <Button
+              label={secondaryButtonLabel}
+              onClick={secondaryButtonFunction}
+              style={{
+                flexGrow: '2',
+                paddingLeft: '70px',
+                paddingRight: '70px',
+                borderRadius: '4px',
+                marginBottom: '40px',
+              }}
+            />
```

Comment:
When I read this code, it makes me feel like this component doesn't know enough about how it will be used.

I think there are two paths here:
1. The component decides what these two buttons mean
2. The consumer decides which buttons, if any, should be shown

For case 1:
You could decide that button one is something like confirm, so it would be `confirmLabel` and `onConfirm`. Button two would be dismiss, so you'd have `dismissLabel` and `onDismiss`. 
That way the props make more sense, and you'd always show two buttons. Or you could add a prop called `dismissible` which controls if the second button is shown.

For case 2:
You could allow the consumer to pass in the buttons altogether, either in a data format or directly as components.

```js
const actions = [
 { label: 'Confirm', callback: () => setModalOpen(false) },
];
<Modal actions={actions} />
```

```js
<Modal>
  <Button label="Confirm" onClick={() => setModalOpen(false)} />
</Modal>
```